great_expectations/core/util.py

```diff
@@ -331,6 +332,9 @@ def convert_to_json_serializable(  # noqa: C901 - complexity 32
     if isinstance(data, bytes):
         return str(data)
 
+    if isinstance(data, pathlib.PurePath):
```

Comment:
All `pathlib` classes inherit from `PurePath` per the diagram at the top of this page: https://docs.python.org/3/library/pathlib.html